vm-setup/roles/v1aX_integration_test/tasks/upgrade.yml

```diff
@@ -1,52 +1,8 @@
 ---
-  - name: Define number of BMH's
-    set_fact:
-      NUMBER_OF_BMH: "{{ NUM_OF_MASTER_REPLICAS|int +  NUM_OF_WORKER_REPLICAS|int }}"
-
-  - name: Update MachineDeployment maxSurge and maxUnavailable fields
-    kubernetes.core.k8s:
-      api_version: cluster.x-k8s.io/{{ CAPI_VERSION }}
-      kind: MachineDeployment
-      name: "{{ CLUSTER_NAME }}"
-      namespace: "{{ NAMESPACE }}"
-      kubeconfig: "/tmp/kubeconfig-{{ CLUSTER_NAME }}.yaml"
-      resource_definition:
-        spec:
-          strategy:
-            rollingUpdate:
-              maxSurge: 1
-              maxUnavailable: 1
-
-  - name: Untaint all nodes
-    ansible.builtin.command: kubectl taint nodes --all node-role.kubernetes.io/master-
-    environment:
-      KUBECONFIG: "/tmp/kubeconfig-{{ CLUSTER_NAME }}.yaml"
-    ignore_errors: yes
```

Comment:
The Untaint here does not provide any useful addition, we have another untaint task after upgrading the controlplane to allow KCP to tolerate worker pods while that single worker is upgrading